src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -144,6 +144,14 @@
         name="warehouse_size",
         type="warehouse_size",
     ),
+    # We use a RegexParser instead of keywords as the arguments are optionally quoted.
+    CompressionType=RegexParser(
+        r"'?AUTO'?|'?GZIP'?|'?BZ2'?|'?BROTLI'?|'?ZSTD'?|'?DEFLATE'?|'?RAW_DEFLATE'?|"
+        r"'?LZO'?|'?NONE'?|'?SNAPPY'?",
+        CodeSegment,
+        name="compression_type",
+        type="compression_type",
```

Comment:
Giving these separate `types` and `names` has positives and negatives.

On one hand, it's good to classify them separately when you can. You could then potentially treat them differently.

However it also means it won't be caught for Keyword capitalisation rules (e.g. L010).

If you want them included in that you could do it like this:

```suggestion
        name="compression_type",
        type="keyword",
```

This also allows you to target it separately (using `name`) but also treat them the same (using `type`) sometimes.

I'm in two minds whether to leave as is, or go with this. As I say pluses and minuses.

WDYT? Are they keywords? Should they be?

Reply:
I spoke to @WittierDinosaur and he also thought `keyword`, done!